src/poly/kzg/mod.rs

```diff
@@ -292,6 +292,8 @@ where
 
         let pi: E::G1Affine = transcript.read().map_err(|_| Error::SamplingError)?;
 
+        transcript.check_empty().map_err(|_| Error::OpeningError)?;
```

Comment:
a few things:

* I would move this at the end of the function, it looks cleaner and you won't have to think about it if you ever have to add more reads after that point in the future
* I would document in both kzg and plonk verifier that the verifier will error if there's trailing bytes in the transcript (as maybe the caller doesn't expect that)
* `read_to_end` reads the whole thing, reading just a byte would be faster